src/Symfony/Component/Translation/Command/XliffLintCommand.php

```diff
@@ -110,22 +112,32 @@ private function validate($content, $file = null)
 
         $document = new \DOMDocument();
         $document->loadXML($content);
-        if ($document->schemaValidate(__DIR__.'/../Resources/schemas/xliff-core-1.2-strict.xsd')) {
-            return array('file' => $file, 'valid' => true);
-        }
 
-        $errorMessages = array_map(function ($error) {
-            return array(
-                'line' => $error->line,
-                'column' => $error->column,
-                'message' => trim($error->message),
+        $expectedFileName = sprintf('messages.%s.xlf', $targetLanguage = $this->getTargetLanguageFromFile($document));
+        $realFileName = basename($file);
+
+        if ($expectedFileName !== $realFileName) {
+            $errors[] = array(
+                'line' => null,
+                'column' => null,
+                'message' => sprintf('There is a mismatch between the file extension ("%s") and the "%s" value used in the "target-language" attribute of the file.', $realFileName, $targetLanguage),
             );
-        }, libxml_get_errors());
+        }
+
+        if (false === $document->schemaValidate(__DIR__.'/../Resources/schemas/xliff-core-1.2-strict.xsd')) {
+            $errors[] = array_map(function ($error) {
```

Comment:
are you sure about it ? This will add an array of errors a a single entry in `$errors`. This looks weird to me

Reply:
btw, your new test does not seem to cover this code path